### PR TITLE
building-comparison: fix checks and result description

### LINKED ISSUE #731: building-comparison: edge case description string shows ratio instead of percent

### LINKED ISSUE #732: building-comparison: check eubucco coverage calculates with sea area
The function `get_eubucco_coverage_intersection_area` compares the area of the eubucco boundaries, which are on the coastline, with the dashboard bpolys, which are the sealine.

### LINKED ISSUE #733: building-comparison: edge case check before requesting data from db?
If the coverage is below 85% the results are not show anyway, so why still requesting the data from the db and calculate it?

### LINKED ISSUE #734: test(building-comparison): mock `get_eubucco_coverage_intersection_area` in `test_building_comparison.py`

### LINKED ISSUE #735: building-comparision: clip AOI

### LINKED ISSUE #737: building-comparison: flag the case when OSM has much more (over 100%) then EUBUCCO
Currently it is green all the way from 100% till infinity.

Lets mark it as undefined and add a warning when completeness is at over 150%

Add additional edge case if over 100 % ?

### PR DESCRIPTION
**Pull request by Gigaszi**
### Description
Please add a clear and concise description of what your PR solves.

### Corresponding issue
Closes #733 
Closes #731 
Closes #735
Closes #732
Closes #737 
Closes #734 

### New or changed dependencies
-

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-api/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-api/blob/main/CHANGELOG.md)

Please check all finished tasks. If some tasks do not apply to your PR, please cross their text out (by using `~...~`) and remove their checkboxes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+
+### New Features
+
+- building-comparison: AOI is now clipped to the coverage area ([#739])
+
+### Other Changes
+
+- building-comparison: no quality estimation for areas with strong difference to reference data ([#739])
+
+
```
Comment (matthiasschaub) [inline CHANGELOG.md:18]:
```suggestion
[#739]: https://github.com/GIScience/ohsome-quality-api/issues/739

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+- building-comparison: is calculated for coverage area only ([#739])
+- building-comparison: result description now shows correct coverage percentage ([#739])
+
+### New Features
+
+- building-comparison: AOI is now clipped to the coverage area ([#739])
+
+### Other Changes
+
+- building-comparison: no quality estimation for areas with strong difference to reference data ([#739])
```
Comment (matthiasschaub) [inline CHANGELOG.md:16]: What about the test fix and the issue number? #734 
Its closes as well right?

File: tests/integrationtests/test_geodatabase.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 import ohsome_quality_api.geodatabase.client as db_client
 
-pytestmark = pytest.mark.skip("dependency on database setup.")
+# pytestmark = pytest.mark.skip("dependency on database setup.")
```
Comment (matthiasschaub) [inline tests/integrationtests/test_geodatabase.py:9]:
```suggestion
pytestmark = pytest.mark.skip("dependency on database setup.")
```

**Review by matthiasschaub — CHANGES_REQUESTED** [review]

**Review by matthiasschaub — APPROVED** [review]
